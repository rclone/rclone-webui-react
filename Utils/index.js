import checkPropTypes from 'check-prop-types';
import {applyMiddleware, createStore} from 'redux';
import rootReducer from './../src/reducers';
import {middleware} from '../src/store';

export const findByTestAttr = (component, attr) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
};

export const checkProps = (component, expectedProps) => {
    const propsErr = checkPropTypes(component.propTypes, expectedProps, 'prop', component.name);
    return propsErr;
};

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};