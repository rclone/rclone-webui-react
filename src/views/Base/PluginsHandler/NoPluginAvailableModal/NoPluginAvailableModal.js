import React from 'react';
import {Button, Modal, ModalBody, ModalFooter, ModalHeader, Table} from "reactstrap";
import * as PropTypes from 'prop-types';

const NoPluginAvailableModal = (props) => {

    const {
        className,
        isOpen,
        setIsOpen,
        availablePlugins,
        processOpenPlugin
    } = props;


    const toggle = () => {
        setIsOpen(!isOpen);
    }


    const openPlugin = (plugin) => {
        processOpenPlugin(plugin.name, plugin.author);
    }

    return (<div data-test="noPluginAvailableModal">
        <Modal isOpen={isOpen} toggle={toggle} className={className}>
            <ModalHeader toggle={setIsOpen}>Couldn't find a plugin to open this file</ModalHeader>
            <ModalBody>
                <Table responsive className="table-striped">
                    <thead>
                    <tr>
                        <th>Name</th>
                        <th>Author</th>
                        <th>Description</th>
                        <th></th>
                    </tr>
                    </thead>
                    <tbody>

                    {
                        !availablePlugins || availablePlugins.length <= 0 ?
                            <p className="p-3">Can't find a plugin to open this file. Go to the store and install
                                some!</p> :
                            availablePlugins.map((value, i) =>
                                <tr key={value.name}>
                                    <td>{value.name}</td>
                                    <td>{value.author}</td>
                                    <td>{value.description}</td>
                                    <td>
                                        <Button color="link" onClick={() => openPlugin(value)}> <i
                                            className="fa fa-lg fa-link"/></Button>
                                    </td>
                                </tr>
                            )}
                    </tbody>
                </Table>

            </ModalBody>
            <ModalFooter>
                <Button data-test="ok-button" color="primary">Go to Store</Button>{' '}
                <Button data-test="cancel-button" color="secondary" onClick={toggle}>Cancel</Button>
            </ModalFooter>
        </Modal>
    </div>);
}

NoPluginAvailableModal.propTypes = {
    className: PropTypes.string.isRequired,
    availablePlugins: PropTypes.array.isRequired,
    isOpen: PropTypes.bool.isRequired,
    setIsOpen: PropTypes.func.isRequired,
    processOpenPlugin: PropTypes.func.isRequired
}

export default NoPluginAvailableModal;