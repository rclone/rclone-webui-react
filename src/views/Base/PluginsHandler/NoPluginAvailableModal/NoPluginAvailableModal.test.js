import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import {findByTestAttr, testStore} from "../../../../../Utils";
import NoPluginAvailableModal from "./NoPluginAvailableModal";


const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);

    const component = shallow(<NoPluginAvailableModal {...props} store={store}/>);
    return component;
};


describe('No plugin available Modal', function () {
    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {
                className: "",
                isOpen: true,
                setIsOpen: jest.fn(),
                availablePlugins: [],
                processOpenPlugin: jest.fn()
            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAttr(wrapper, "noPluginAvailableModal");
            expect(component).toHaveLength(1);
        });

        it('should have ok and cancel buttons', function () {
            const okButton = findByTestAttr(wrapper, "ok-button")
            expect(okButton).toHaveLength(1);

            const cancelButton = findByTestAttr(wrapper, "cancel-button");
            expect(cancelButton).toHaveLength(1);
        })


        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

    });
});
