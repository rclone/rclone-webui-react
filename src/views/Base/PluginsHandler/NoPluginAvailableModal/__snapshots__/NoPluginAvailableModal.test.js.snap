// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No plugin available Modal renders should match snapshot 1`] = `
<div
  data-test="noPluginAvailableModal"
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    className=""
    container="body"
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      toggle={[MockFunction]}
      wrapTag="div"
    >
      Couldn't find a plugin to open this file
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <Table
        className="table-striped"
        responsive={true}
        responsiveTag="div"
        tag="table"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Author
            </th>
            <th>
              Description
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <p
            className="p-3"
          >
            Can't find a plugin to open this file. Go to the store and install some!
          </p>
        </tbody>
      </Table>
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="primary"
        data-test="ok-button"
        tag="button"
      >
        Go to Store
      </Button>
       
      <Button
        color="secondary"
        data-test="cancel-button"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
